7/Dockerfile

```diff
@@ -1,46 +1,43 @@
-FROM centos:centos7
+FROM centos/s2i-base-centos7
 
-MAINTAINER Tom Stellard <tstellar@redhat.com>
+ENV LLVM_VERSION="4.0.x" \
+    SUMMARY="Platform for building projects with clang/llvm" \
+    DESCRIPTION="$SUMMARY"
 
-ENV SUMMARY="Red Hat LLVM Toolset 7 container image" \
-    DESCRIPTION="Platform for building applications using Red Hat LLVM Toolset 7"
 
-LABEL com.redhat.component="llvm-toolset-7-docker" \
-      name="devtools/llvm-toolset-7-rhel7" \
-      version="7" \
-      summary="$SUMMARY" \
+LABEL summary="$SUMMARY" \
       description="$DESCRIPTION" \
       io.k8s.description="$DESCRIPTION" \
-      io.k8s.display-name="LLVM Toolset 7"
-
-RUN yum install -y yum-utils && \
-    yum install -y centos-release-scl && \
+      io.k8s.display-name="$LLVM_VERSION" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.tags="builder,llvm,clang" \
+      com.redhat.component="llvm-toolset-7-docker" \
+      name="centos/llvm-toolset-7-centos7" \
+      version="1" \
+      maintainer="SoftwareCollections.org <sclorg@redhat.com>"
+
+RUN yum install -y centos-release-scl-rh && \
     yum-config-manager --enable centos-sclo-rh-testing && \
-    INSTALL_PKGS="llvm-toolset-7" && \
+    INSTALL_PKGS="llvm-toolset-7 llvm-toolset-7-cmake" && \
     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all -y
 
-COPY ./root /
-
-ENV HOME=/opt/app-root/src \
-    PATH=/opt/app-root/src/bin:/opt/app-root/bin:/opt/rh/llvm-toolset-7/root/usr/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+COPY ./s2i/bin/ $STI_SCRIPTS_PATH
 
-RUN mkdir -p ${HOME} && \
-    groupadd -r default -f -g 1001 && \
-    useradd -u 1001 -r -g default -d ${HOME} -s /sbin/nologin \
-    -c "Default Application User" default && \
-    chown -R 1001:1001 /opt/app-root /usr/bin/container-entrypoint /usr/bin/usage && \
-    chmod u+x /usr/bin/usage
-
-USER 1001
-
-WORKDIR ${HOME}
+# Drop the root user and make the content of /opt/app-root owned by user 1001
+RUN chown -R 1001:0 /opt/app-root && \
+    chmod -R ug+rwx /opt/app-root && \
+    rpm-file-permissions
 
 # Enable the SCL for all bash scripts.
 ENV BASH_ENV=/opt/app-root/etc/scl_enable \
     ENV=/opt/app-root/etc/scl_enable \
-    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable"
+    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable" \
+    PATH=/opt/rh/llvm-toolset-7/root/usr/bin:$PATH \
```

Comment:
Will this work?

`scl enable` does also other things than changing PATH. So for example LD_LIBRARY_PATH is set wrong for binaries in `/opt/rh/llvm-toolset-7/root/usr/bin`

Reply:
Should I just drop PATH= part ?

Reply:
@tstellar Yes, I think it's better to drop it, since in most cases the trick with `PROMPT_COMMAND`, `ENV`, and `BASH_ENV` variables should result in the `llvm-toolset-7` SCL to be enabled every time the docker daemon is started.

Reply:
@hhorak I've done a little more testing, and it looks the trick with PROMPT_COMMAND, ENV, and BASH_ENV does not enable the llvm-toolset-7 SCL.  Looking at the contents of /opt/app-root/etc/scl_enable, all I see is:

# This file contains automatic SCL enablement.
unset BASH_ENV PROMPT_COMMAND ENV

Is there some other command I'm missing that is supposed to populate this file with something related to llvm-toolset-7?

Reply:
It turns out this was because we weren't copying the scl_enable file to the container.